### PR TITLE
Bug 1809187: OpenStack: return an error if load balancer deletion fails

### PR DESCRIPTION
**Pull request by Fedosin**
Now if there was a failure during load balancer deletion, we do not return anything and continue the execution, which leads to a runtime error later.
This commit changes this behavior and returns an error immediately if something bad happens.

### CODE REVIEW

File: pkg/destroy/openstack/openstack.go [hunk 1]
```diff
@@ -671,18 +671,25 @@ func deleteLoadBalancers(opts *clientconfig.ClientOpts, filter Filter, logger lo
...
 	conn, err := clientconfig.NewServiceClient("load-balancer", opts)
 	if err != nil {
-		logger.Debugf("%v", err)
-		return true, nil
+		// Ignore the error if Octavia is not available for the cloud
+		if _, ok := err.(*gophercloud.ErrEndpointNotFound); ok {
+			logger.Debug("Skip load balancer deletion because Octavia endpoint is not found")
+			return true, nil
+		}
+		logger.Errorf("%v", err)
```
Comment (abhinavdahiya) [inline pkg/destroy/openstack/openstack.go]: why not just `logger.Error` ?

Comment (Fedosin) [reply pkg/destroy/openstack/openstack.go]: This pattern is used everywhere in the file and I decided to keep it for consistency

**Comment by mandre** [comment]
/lgtm
@abhinavdahiya let's fix the use of `logger.Errorf` in a follow-up commit since it's everywhere in this file.

**Comment by mandre** [comment]
> /lgtm
> @abhinavdahiya let's fix the use of `logger.Errorf` in a follow-up commit since it's everywhere in this file.

Done in https://github.com/openshift/installer/pull/3219

**Pull request merged**